konnectivity-client/pkg/client/client.go

```diff
@@ -501,11 +501,16 @@ func (t *grpcTunnel) Recv() (*client.Packet, error) {
 	t.recvLock.Lock()
 	defer t.recvLock.Unlock()
 
+	const segment = commonmetrics.SegmentToClient
 	pkt, err := t.stream.Recv()
 	if err != nil && err != io.EOF {
-		metrics.Metrics.ObserveStreamErrorNoPacket(commonmetrics.SegmentToClient, err)
+		metrics.Metrics.ObserveStreamErrorNoPacket(segment, err)
 	}
-	return pkt, err
+	if err != nil {
```

Comment:
This makes me wonder whether we should record packets as sent if there's a stream error...

Reply:
I understand your point, but to make comparisons (sent vs received) I prefer the meaning "desired" over "actual", and doing so would lose the former.